imagej/imagej-ops2/src/main/java/net/imagej/ops2/filter/convolve/ConvolveFFTF.java

```diff
@@ -83,19 +84,20 @@
 	 *
 	 * @param input
 	 * @param kernel
-	 * @param borderSize
-	 * @param obfInput
 	 * @param outType
-	 * @param fftType
+	 * @param complexType
 	 * @param executorService
+	 * @param borderSize (required = false)
+	 * @param obfInput (required = false)
 	 * @return the output
 	 */
 	@Override
 	public RandomAccessibleInterval<O> apply(final RandomAccessibleInterval<I> input,
-			final RandomAccessibleInterval<K> kernel, final long[] borderSize,
-			final OutOfBoundsFactory<I, RandomAccessibleInterval<I>> obfInput,
+			final RandomAccessibleInterval<K> kernel,
 			final O outType, final C complexType,
-			final ExecutorService es) {
+			final ExecutorService executorService,
+			@Optional final long[] borderSize,
+			@Optional OutOfBoundsFactory<I, RandomAccessibleInterval<I>> obfInput) {
```

Comment:
I know that the Op that uses this parameter does a `null` check, but we can't assume that will always be the case - someone could write another Op that gets matched instead, where this OOBF is not required. It's probably not going to happen here, but I don't want to get in the habit of doing this.

I think we have to require that any Op declaring Optional parameters **has** to populate the nulls - it can't let them propagate down into dependencies.